### DISCUSSION TITLE
Remove vertices with no associated triangle(s)

### DISCUSSION THREAD

**Discussion by nebis-software** [donmccurdy/glTF-Transform#988] [Q&A] [answered]
Hi,
    I am using glTF-Transform CLI to cleanup/optimize models and so far it is working great :)
I have noticed that some models have vertices with NO associated triangle: i would like to remove that vertices since they are not very useful for my application.

Do you know if this can be done? 
Thanks!
Luigi

PS: I am mostly using glb models

**Comment by donmccurdy** [comment, marked as answer]
Hi @nebis-software! This is not officially a "feature" of the CLI, but I think you could still remove unused vertices by first un-welding and then re-welding meshes:

```
gltf-transform unweld in.glb tmp.glb
gltf-transform weld tmp.glb out.glb
```

If the model doesn't look correct after welding, you may need to reduce the tolerance for the weld step with a flag like:

 `--tolerance 0.00001`

**Reply by nebis-software** [reply]
I will check, thanks!

**Comment by georgemalick412** [comment]
Hi Luigi,
I'm glad to hear that the glTF-Transform CLI is working well for you. To remove vertices with no associated triangles, you can try using the "prune" operation in glTF-Transform. This operation helps remove unused elements from the model [Welding Trend](https://weldingtrends.com/), including vertices without triangles. Since you're primarily working with glb models, the "glTF-Transform prune" command should help you achieve the desired result. Give it a try and let me know if it works for you.

**Reply by donmccurdy** [reply]
Hi @georgemalick412, unfortunately this answer is factually incorrect. If you're writing responses using LLM-based software, as appears to be the case, please manually verify that the answer works before sharing it.

**Comment by kzhsw** [comment]
Seems that [reindex](https://github.com/donmccurdy/glTF-Transform/discussions/927) could work for this case, but unfortunately this is not included in the cli, coding in js or ts is needed to use it in your project.
